core/src/test/java/it/pagopa/selfcare/mscore/core/DelegationServiceImplTest.java

```diff
@@ -45,47 +45,70 @@ class DelegationServiceImplTest {
     private static final Delegation dummyDelegationProdIo;
     private static final Delegation dummyDelegationProdPagopa;
 
+    private static final Delegation dummyDelegationTaxCode;
+
     static {
         dummyDelegationProdIo = new Delegation();
         dummyDelegationProdIo.setId("id");
+        dummyDelegationProdIo.setTo("to");
+        dummyDelegationProdIo.setFrom("from");
         dummyDelegationProdIo.setProductId("prod-io");
 
         dummyDelegationProdPagopa = new Delegation();
         dummyDelegationProdPagopa.setId("id");
         dummyDelegationProdPagopa.setProductId("prod-pagopa");
-        dummyDelegationProdPagopa.setTo("to");
+        dummyDelegationProdPagopa.setTo("taxCodeTo");
         dummyDelegationProdPagopa.setFrom("from");
+
+        dummyDelegationTaxCode = new Delegation();
+        dummyDelegationTaxCode.setToTaxCode("taxCodeTo");
+        dummyDelegationTaxCode.setFromTaxCode("taxCodeFrom");
+        dummyDelegationTaxCode.setProductId("prod-pagopa");
+        dummyDelegationTaxCode.setId("id");
     }
 
     /**
      * Method under test: {@link DelegationServiceImpl#createDelegation(Delegation)}
      */
     @Test
     void testCreateDelegationWithProductProdIo() {
-        Institution institution = new Institution();
-        institution.setId("id");
+        Institution institutionTo = new Institution();
+        institutionTo.setId("idTo");
+        institutionTo.setTaxCode("taxCodeTo");
+        Institution institutionFrom = new Institution();
+        institutionTo.setId("idFrom");
+        institutionTo.setTaxCode("taxCodeFrom");
         when(delegationConnector.save(any())).thenReturn(dummyDelegationProdIo);
```

Comment:
you could user .thenAnswer method instead of .thenReturn, for ex.` .thenAnswer(arg -> return (Delegation) arg.getArguments()[0])`